### PR TITLE
Fix show grants does not honor given schema name

### PR DESCRIPTION
**Pull request by HiramJ**
This is a fix for #8204.
Added an additional filter on schema name when the table name is given.

### CODE REVIEW

**Comment by HiramJ** [comment]
Thanks @amrutagokhale, I've changed the commit message.

File: presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java [hunk 1]
```diff
@@ -226,7 +226,13 @@ protected Node visitShowGrants(ShowGrants showGrants, Void context)
                         session.getIdentity(),
                         new CatalogSchemaName(catalogName, qualifiedTableName.getSchemaName()));
 
-                predicate = Optional.of(equal(identifier("table_name"), new StringLiteral(qualifiedTableName.getObjectName())));
+                String schemaName = qualifiedTableName.getSchemaName();
+                predicate = Optional.of(
+                        logicalAnd(
+                                equal(identifier("table_schema"), new StringLiteral(schemaName)),
+                                equal(identifier("table_name"), new StringLiteral(qualifiedTableName.getObjectName()))
+                        )
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java]: Put both of these parenthesis on the previous line

File: presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java [hunk 2]
```diff
@@ -226,7 +226,13 @@ protected Node visitShowGrants(ShowGrants showGrants, Void context)
                         session.getIdentity(),
                         new CatalogSchemaName(catalogName, qualifiedTableName.getSchemaName()));
 
-                predicate = Optional.of(equal(identifier("table_name"), new StringLiteral(qualifiedTableName.getObjectName())));
+                String schemaName = qualifiedTableName.getSchemaName();
+                predicate = Optional.of(
+                        logicalAnd(
```
Comment (electrum) [inline presto-main/src/main/java/com/facebook/presto/sql/rewrite/ShowQueriesRewrite.java]: Put this on previousl ine

**Comment by HiramJ** [comment]
Thanks @electrum, I've fixed the coding style and also added a test for the change.

**Pull request closed without merging**